dcp_prototype/backend/browser/api/app.py

```diff
@@ -0,0 +1,151 @@
+
+from browser.rds.browser_orm import (DBSessionMaker,
+                                     Project, File,
+                                     LibraryPrepProtocol, Tissue, Species,
+                                     LibraryPrepProtocolJoinProject, TissueJoinProject,
+                                     SpeciesJoinProject)
+
+app = Chalice(app_name='browser-api')
+
+
+@app.route('/')
+def index():
+    return "render documentation here"
+
+
+@app.route('/projects')
+def get_projects():
+    session = DBSessionMaker().session()
+    projects = []
+    for project in session.query(Project).all():
+        projects.append({
+            'id': project.id,
+            'title': project.title,
+            'assays': _get_project_assays(session, project.id),
+            'tissues': _get_project_tissues(session, project.id),
+            'species': _get_project_species(session, project.id),
+            'cell_count': project.cell_count,
+        })
+
+    return chalice.Response(status_code=200,
+                            headers={
+                                'Content-Type': "application/json",
+                                'Access-Control-Allow-Origin': "*"
+                            },
+                            body=projects)
+
+
+@app.route('/projects/{project_id}')
+def get_project(project_id):
+    session = DBSessionMaker().session()
+    project = session.query(Project).get(project_id)
+    response_body = {
+        'id': project.id,
+        'title': project.title,
+        'label': project.label,
+        'assays': _get_project_assays(session, project.id),
+        'tissues': _get_project_tissues(session, project.id),
+        'species': _get_project_species(session, project.id),
+        'description': project.description,
+        'category': project.category,
+        'developmental_stage': project.developmental_stage,
+        'disease_ontology': project.disease_ontology,
+        'sample_type': project.sample_type,
+        'organ_part': project.organ_part,
+        'analysis_protocol': project.analysis_protocol.split(','),
+        'cell_count': project.cell_count,
+        'donor_count': project.donor_count,
+        'publication_title': project.publication_title,
+        'publication_doi': project.publication_doi,
+        'contact_name': project.contact_name,
+        'contact_institution': project.contact_institution,
+        'contact_email': project.contact_email
+    }
+
+    return chalice.Response(status_code=200,
+                            headers={
+                                'Content-Type': "application/json",
+                                'Access-Control-Allow-Origin': "*"
+                            },
+                            body=response_body)
+
+
+@app.route('/projects/{project_id}/files')
+def get_project_files(project_id):
+    session = DBSessionMaker().session()
+    files = []
+    for file in session.query(File).filter(File.project_id == project_id).limit(100):
```

Comment:
Is there already a ticket created to handle pagination?

Reply:
_tl;dr I'm proposing this endpoint return only metadata on downloadable files for which pagination is not required._

I'm thinking we won't need pagination for the foreseeable future on the endpoint. There are two sets of files the backend is interested in returning metadata for:

1. Downloadable files (grouped data, metadata, matrices)
1. Individual files, i.e. content of metadata in [1]

The `/projects{project_id}/files` endpoint is concerned with [1] for which the number of files should be manageable and not require pagination.

[2] will be handled by the Artifact -> Browser ETL for which the files table can be queried directly.

Reply:
To your q, here's the [ticket](https://app.zenhub.com/workspaces/dcp2-5e2a191dad828d52cc78b028/issues/humancellatlas/dcp-prototype/130) to implement this endpoint